### ISSUE TITLE
Not able to set AAM refreshable tokens

### ISSUE THREAD

**Issue by maares** [aamplugin/advanced-access-manager#145]
Hi,
trying to create tokens refreshable as default

- Added to configPress the following line

`authentication.jwt.refreshable = true` or `authentication.jwt.refreshableJWT = true`

_Did not work_

- Adding to the Post creation of the token the following line

```
{
	"username": "$user",
	"password": "$pass",
	"issueJWT": true,
        "refreshableJWT": true
}
```

_Do work_

As administrator I can generate a refreshable Token, on the contrary as a normal user I can not. 
```
{
    "reason": "Current user is not allowed to issue refreshable JWT token"
}
```

Could not find any option on AAM console which configures this.

Would appreciate some guidance.

Non related questions, on one of your tuto videos, I could see Social Login integration but could not find it on AAM dashboard from WP. Have they been removed?

Thanks in advance!

**Comment by maares** [comment]
Hi, any hint on how to configure AAM?

**Comment by aamplugin** [comment]
@maares oh, I see. Well, neither methods are valid. In order to allow a user to be able to generate refreshable token, you need to grant the custom capability `aam_issue_refreshable_jwt`. This way, any user who has this capability can send `refreshableJWT` in the payload and receive back the refreshable token.

Keep me posted if anything.

**Comment by kveres0** [comment]
Hi,
I configured the aam_issue_refreshable_jwt capability and everything worked as expected, but with the 5.5.1 wordpress update i got the error message if i want to get refreshable token. The normal one is fine.
I had an old wp backup and i checked it: in that version everything was ok.

**Comment by maares** [comment]
@aamplugin thanks for the reply. Trying to get it working I even have all permissions levels to an specific user and still could not get it working. Went through your doc https://aamplugin.com/reference/plugin#aam_issue_refreshable_jwt is still missing.

Could you post how to configure aam_issue_refreshable_jwt capability to create refreshable tokens by default to specific roles or even all?

**Comment by kveres0** [comment]
Working with administrative role, but not with anything else

**Comment by maares** [comment]
@kveres0 maybe I'm wrong, but are you telling me that not other role than admin can generate refreshable tokens?

**Comment by kveres0** [comment]
Yes. A few week ago (as i mentioned, maybe with the 5.5.1 wp update) only the admin can generate these tokens. I checked with postman several roles, without refreshableJWT option everything is working, with it (in my site) only the admin.
I found a backup with 6.6.2 (maybe on wp 5.5) and it was good. If it helps, tomorrow i can try to find it again.

**Comment by maares** [comment]
@kveres0 many thanks for your feedback, it's a shame as the main reason to migrate to AAM was the refresh functionality.

Once again thanks

**Comment by Madaxen** [comment]
> @kveres0 many thanks for your feedback, it's a shame as the main reason to migrate to AAM was the refresh functionality.
> 
> Once again thanks

I got it working. What you have to do is what aamplugin said with _"to grant the custom capability aam_issue_refreshable_jwt"_ you can do that like so in your wordpress account. You may have to add this action to all roles you require:
![grafik](https://user-images.githubusercontent.com/73057544/96363289-9bcf7200-1133-11eb-8bfc-8db6ca865af3.png)

**Comment by kveres0** [comment]
thanks Madaxen,
I gave this capability to the user (without success), but (as you) after assigning to the role, it's working. This is the magic. :)

**Comment by madexen86** [comment]
@maares : Please check if the solution is working for you and close the issue.
